code/modules/cooking/machinery/smartfridge.dm

```diff
@@ -236,8 +210,14 @@
 	return
 
 /obj/machinery/smartfridge/machinery_process()
-	if(stat & (BROKEN|NOPOWER))
+	update_icon()
```

Comment:
I think you could just move the two update_icon() procs into update_use_power instead.

Reply:
Good call. Done.